contracts/staking/contracts/src/vaults/StakingPoolRewardVault.sol

```diff
@@ -172,35 +173,78 @@ contract StakingPoolRewardVault is
     /// Note that this is only callable by the staking contract, and when
     /// not in catastrophic failure mode.
     /// @param poolId Unique Id of pool.
-    /// @param poolOperatorShare Fraction of rewards given to the pool operator, in ppm.
-    function registerStakingPool(bytes32 poolId, uint32 poolOperatorShare)
+    /// @param operatorShare Fraction of rewards given to the pool operator, in ppm.
+    function registerStakingPool(
+        bytes32 poolId,
+        address payable operatorAddress,
+        uint32 operatorShare
+    )
         external
         onlyStakingContract
         onlyNotInCatastrophicFailure
     {
         // operator share must be a valid fraction
-        if (poolOperatorShare > PPM_DENOMINATOR) {
-            LibRichErrors.rrevert(LibStakingRichErrors.InvalidPoolOperatorShareError(
+        if (operatorShare > PPM_DENOMINATOR) {
+            LibRichErrors.rrevert(LibStakingRichErrors.OperatorShareError(
+                LibStakingRichErrors.OperatorShareErrorCodes.OPERATOR_SHARE_MUST_BE_BETWEEN_0_AND_100,
                 poolId,
-                poolOperatorShare
+                operatorShare
             ));
         }
 
         // pool must not exist
-        Balance memory balance = balanceByPoolId[poolId];
-        if (balance.initialized) {
+        Pool memory pool = poolById[poolId];
+        if (pool.initialized) {
             LibRichErrors.rrevert(LibStakingRichErrors.PoolAlreadyExistsError(
                 poolId
             ));
         }
 
-        // set initial balance
-        balance.initialized = true;
-        balance.operatorShare = poolOperatorShare;
-        balanceByPoolId[poolId] = balance;
+        // initialize pool
+        pool.initialized = true;
+        pool.operatorAddress = operatorAddress;
+        pool.operatorShare = operatorShare;
+        poolById[poolId] = pool;
 
         // notify
-        emit StakingPoolRegistered(poolId, poolOperatorShare);
+        emit StakingPoolRegistered(poolId, operatorShare);
+    }
+
+    /// @dev Decreases the operator share for the given pool (i.e. increases pool rewards for members).
+    /// Note that this is only callable by the staking contract, and will revert if the new operator
+    /// share value is greater than the old value.
+    /// @param poolId Unique Id of pool.
+    /// @param newOperatorShare The newly decreased percentage of any rewards owned by the operator.
+    function decreaseOperatorShare(bytes32 poolId, uint32 newOperatorShare)
+        external
+        onlyStakingContract
+        onlyNotInCatastrophicFailure
+    {
+        Pool memory pool = poolById[poolId];
+        uint32 oldOperatorShare = pool.operatorShare;
+
+        if (newOperatorShare > oldOperatorShare) {
```

Comment:
I wonder if it makes sense to just revert if `newOperatorShare == oldOperatorShare`. Otherwise, it could be an expensive operation that doesn't do anything. Also, the event would be a little confusing in this case. 

Thoughts?

Reply:
makes sense to me